### PR TITLE
Do not add sourcemap markings for indentation

### PR DESCRIPTION
**Pull request by jridgewell**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            |
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

This part 1 of me drastically simplifying the sourcemap generation now that we're off the `source-map` library.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/14506"></a>

### CODE REVIEW

File: packages/babel-core/test/fixtures/transformation/source-maps/full/source-map.json [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
-  "mappings": "AAAAA,GAAG,CAACC,GAAJ,CAAQ,UAAAC,CAAC;AAAA,SAAIA,CAAC,GAAGA,CAAR;AAAA,CAAT",
-  "sourcesContent": ["arr.map(x => x * x);"]
-}
+  "sources": [
+    "source-maps/full/input.js"
+  ],
+  "sourcesContent": [
+    "arr.map(x => x * x);"
+  ],
+  "mappings": "AAAAA,GAAG,CAACC,GAAJ,CAAQ,UAAAC,CAAC;EAAA,OAAIA,CAAC,GAAGA,CAAR;AAAA,CAAT"
```
Comment (jridgewell) [inline packages/babel-core/test/fixtures/transformation/source-maps/full/source-map.json:14]: The diff here:
```diff
- AAAAA,GAAG,CAACC,GAAJ,CAAQ,UAAAC,CAAC;AAAA,SAAIA,CAAC,GAAGA,CAAR;AAAA,CAAT
+ AAAAA,GAAG,CAACC,GAAJ,CAAQ,UAAAC,CAAC;EAAA,OAAIA,CAAC,GAAGA,CAAR;AAAA,CAAT
```

The changed `AAAA,SAAIA` -> `EAAA,OAAIA` is the two spaces used for indentation. `A` means 0, and `E` means , so we moved the generated column 2 times when inserting the new marking. The `S` -> `O` is because markings are deltas, so we subtracted two columns from `S` gives us `O`.

File: packages/babel-generator/src/printer.ts [hunk 2]
```diff
@@ -273,7 +273,7 @@ class Printer {
       this.endsWith(charCodes.lineFeed) &&
       str.charCodeAt(0) !== charCodes.lineFeed
     ) {
-      this._buf.queue(this._getIndent());
+      this._buf.queueIndentation(this._getIndent());
```
Comment (jridgewell) [inline packages/babel-generator/src/printer.ts:276]: This is the real change.

File: packages/babel-generator/src/buffer.ts [hunk 3]
```diff
@@ -96,15 +109,22 @@ export default class Buffer {
     this._queue.unshift([str, line, column, identifierName, filename, force]);
   }
 
+  /**
+   * Same as queue, but this indentation will never have a sourcmap marker.
+   */
+  queueIndentation(str: string): void {
```
Comment (jridgewell) [inline packages/babel-generator/src/buffer.ts:115]: This is the real change.

File: packages/babel-generator/src/buffer.ts [hunk 4]
```diff
@@ -2,37 +2,51 @@ import type SourceMap from "./source-map";
...
+}
+
 const SPACES_RE = /^[ \t]+$/;
 export default class Buffer {
   constructor(map?: SourceMap | null) {
     this._map = map;
   }
 
   _map: SourceMap = null;
-  _buf: string = "";
```
Comment (jridgewell) [inline packages/babel-generator/src/buffer.ts:12]: Sorry, but this file needs a lot of cleanup.

File: packages/babel-generator/test/index.js [hunk 5]
```diff
@@ -103,12 +102,6 @@ describe("generation", function () {
           source: "a.js",
           original: { line: 1, column: 18 },
         },
-        {
-          name: "console",
-          generated: { line: 2, column: 0 },
-          source: "a.js",
-          original: { line: 1, column: 20 },
-        },
```
Comment (nicolo-ribaudo) [inline packages/babel-generator/test/index.js:111]: Q: why was this deleted?

**Review by nicolo-ribaudo — APPROVED** [review]

**Review by existentialism — APPROVED** [review]

**Comment by nicolo-ribaudo** [comment]
Thanks!
